signer/src/test/kotlin/io/provenance/hdwallet/signer/TestECDSASignature.kt

```diff
@@ -96,25 +126,102 @@ class TestItAll {
 
                 assertEquals(ekp.expectedRoot.r, rootSig.r, "${test.javaClass.simpleName} $name root r")
                 assertEquals(ekp.expectedRoot.s, rootSig.s, "${test.javaClass.simpleName} $name root s")
-                assertEquals(ekp.expectedRoot.btc.base16Encode(), rootSig.encodeAsBTC().base16Encode(), "${test.javaClass.simpleName} $name root btc")
-                assertEquals(true, signer.verify(keys.root.keyPair.publicKey, payloadHash, rootSig), "${test.javaClass.simpleName} $name root verify")
+                assertEquals(
+                    ekp.expectedRoot.btc.base16Encode(),
+                    rootSig.encodeAsBTC().base16Encode(),
+                    "${test.javaClass.simpleName} $name root btc"
+                )
+                assertEquals(
+                    true,
+                    signer.verify(keys.root.keyPair.publicKey, payloadHash, rootSig),
+                    "${test.javaClass.simpleName} $name root verify"
+                )
 
                 assertEquals(ekp.expectedTestnet.r, testnetSig.r, "${test.javaClass.simpleName} $name test r")
                 assertEquals(ekp.expectedTestnet.s, testnetSig.s, "${test.javaClass.simpleName} $name test s")
-                assertEquals(ekp.expectedTestnet.btc.base16Encode(), testnetSig.encodeAsBTC().base16Encode(), "${test.javaClass.simpleName} $name test btc")
-                assertEquals(true, signer.verify(keys.testnet.keyPair.publicKey, payloadHash, testnetSig), "${test.javaClass.simpleName} $name test verify")
+                assertEquals(
+                    ekp.expectedTestnet.btc.base16Encode(),
+                    testnetSig.encodeAsBTC().base16Encode(),
+                    "${test.javaClass.simpleName} $name test btc"
+                )
+                assertEquals(
+                    true,
+                    signer.verify(keys.testnet.keyPair.publicKey, payloadHash, testnetSig),
+                    "${test.javaClass.simpleName} $name test verify"
+                )
+
+                assertEquals(ekp.expectedMainnet.r, mainnetSig.r, "${test.javaClass.simpleName} $name main r")
+                assertEquals(ekp.expectedMainnet.s, mainnetSig.s, "${test.javaClass.simpleName} $name main s")
+                assertEquals(
+                    ekp.expectedMainnet.btc.base16Encode(),
+                    mainnetSig.encodeAsBTC().base16Encode(),
+                    "${test.javaClass.simpleName} $name main btc",
+                )
+                assertEquals(
+                    true,
+                    signer.verify(keys.mainnet.keyPair.publicKey, payloadHash, mainnetSig),
+                    "${test.javaClass.simpleName} $name main verify"
+                )
+            }
+        }
+    }
+
+    @Test
+    fun testSignatureEncoding() {
+        val keyPair = createECKeyPair()
+        val publicKey = keyPair.first
+        val privateKey = keyPair.second
+
+        fun JCESignature(): Signature = Signature.getInstance("SHA256withECDSA")
```

Comment:
I think you need to use "ECDDSA" instead of "ECDSA" here to get the deterministic signatures.

Reply:
@mtps That's a great point. I was wondering that and was tempted to ask you. With ECDSA, the original key and the converted key are not equal, but they still work for verification--so in that sense they're functionally equivalent. I read up a bit (can't find the resource) that mentioned the presence of randomness/nondeterminism in EDCSA. I'll swap out `ECDSA` with `ECDDSA` and change the tests to enforce post-conversion key equality